build/Common.props

```diff
@@ -28,4 +28,8 @@
     <PackageReference Include="System.Collections.Immutable" Version="[1.7.1,)" />
     <PackageReference Include="System.Threading.Channels" Version="[6.0.0,)" />
   </ItemGroup>
+
+  <ItemGroup Condition="'$(OS)' == 'Unix'">
+    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" PrivateAssets="all" />
+  </ItemGroup>
```

Comment:
> Is this because we're not building on Windows in the job?

Yep! It gives `dotnet build` enough context to build the `net462` TFM on Linux, so we can run package builds on `ubuntu-latest` which, among other reasons, runs circles around `windows-latest` for `dotnet restore` and anything else that uses compression:
- `windows-latest`:
  ```
  Run dotnet test test\OpenFeature.Tests\ --configuration Release --logger GitHubActions
    Determining projects to restore...
    Restored D:\a\dotnet-sdk\dotnet-sdk\src\OpenFeature\OpenFeature.csproj (in 21.2 sec).
    Restored D:\a\dotnet-sdk\dotnet-sdk\test\OpenFeature.Tests\OpenFeature.Tests.csproj (in 1.69 min).
  ```
- `ubuntu-latest`:
  ```
  Run dotnet test test/OpenFeature.Tests/ --configuration Release --logger GitHubActions
    Determining projects to restore...
    Restored /home/runner/work/dotnet-sdk/dotnet-sdk/src/OpenFeature/OpenFeature.csproj (in 5.33 sec).
    Restored /home/runner/work/dotnet-sdk/dotnet-sdk/test/OpenFeature.Tests/OpenFeature.Tests.csproj (in 5.5 sec).
  ```


> Will this have any impact on the release build ([it does run on windows](https://github.com/open-feature/dotnet-sdk/blob/main/.github/workflows/release.yml))?

It should have no impact since it's conditioned on the `OS` var and the reference assemblies won't be included in the compilation, leaving `dotnet` to find the actual assemblies in the GAC (or on disk, can't remember which it actually checks, so don't quote me on this part).

But it does mean, that once everyone is comfortable with it (i.e. this workflow runs without issues a few times), we could then update `release.yml` to run on `ubuntu-latest` too.

That workflow obviously runs less frequently than `ci.yml`, so the `dotnet restore` perf won't be as big of a consideration in that case, but I personally feel better building my internal stable releases on the `ubuntu-latest` runners. Things just always seem to work better there lol (e.g. cli perf, line endings, etc).

Reply:
Makes sense. I'm in favor of that change in the future. Thanks for the explanation 🙏